### ISSUE TITLE
The SCIMMY attributes throw away null value, but for PatchOp can search null value attributes, which makes the filter never match

### ISSUE THREAD

**Issue by jiaoyang28** [scimmyjs/scimmy#34]
Still for the Entra integration, to add the member to a group, the request body looks like (refer to [this link](https://learn.microsoft.com/en-us/entra/identity/app-provisioning/use-scim-to-provision-users-and-groups#update-group-add-members)):
```json
{
    "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
    "Operations": [{
        "op": "Add",
        "path": "members",
        "value": [{
            "$ref": null,
            "value": "f648f8d5ea4e4cd38e9c"
        }]
    }]
}
```

As a result to this [SCIMMY code](https://github.com/scimmyjs/scimmy/blob/18a96c4806dc1f685e48812ec42b3e9bfbdeffdd/src/lib/types/attribute.js#L440), the $ref part is thrown away and never reach to the ingress handler.

However, when remove group member with request body looks like (refer to [this link](https://learn.microsoft.com/en-us/entra/identity/app-provisioning/use-scim-to-provision-users-and-groups#update-group-remove-members)):
```json
{
    "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
    "Operations": [{
        "op": "Remove",
        "path": "members",
        "value": [{
            "$ref": null,
            "value": "f648f8d5ea4e4cd38e9c"
        }]
    }]
}
```

Except the SCIMMY throws the issue mentioned in https://github.com/scimmyjs/scimmy/issues/33, it also fails the Filter match, as the $ref does not exist.

**Comment by LesterLian** [comment]
I think it's fine that `null` value doesn't get through the ingress and stored, but we should apply the same logic when patching. Since the object coming out of egress won't have any `null` value, we should treat `null` value in filters as `undefined`. As suggested in this commit: https://github.com/LesterLian/scimmy/commit/869b405ce7cca50d00672fc1a56247070377c199

Also, I found that leading numbers in a value of the Remove Op will break the filter, since the expression will be something like `value eq 123abc`. I propose to wrap the value in double quotes when the value is string, as in this commit: https://github.com/LesterLian/scimmy/commit/e9c5a46ab3b6e7f3f01529c490404184a0581271

**Comment by sleelin** [comment]
Hi @jiaoyang28, thanks for the feedback!

As per the SCIM protocol specification in [RFC7643§2.5 Unassigned and Null Values](https://datatracker.ietf.org/doc/html/rfc7643#section-2.5), undefined and null are both treated as unassigned values, and attributes whose value is null are thus undefined on resource instances passed to the ingress handler. In the example add operation, the `$ref` attribute with value `null` is superfluous, as it behaves the same as if the attribute had not been specified at all in the operation.

You have still identified a bug, however. Whilst null should be treated as undefined for attribute values, when used in a filter expression, the current matching function is strictly looking for the value `null`, which is incorrect behaviour. Similar to @LesterLian's suggestion, I have modified the matching function so that when an equality comparison (`eq`, `ne`) specifies `null` as the value, it will correctly match objects where the value is undefined, as expected.

The fix for this issue will be included in release v1.2.3.

Thanks,
Sam
